From 66c6f1c284e9b11a6cc5832097eb1374966aa6bf Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Mon, 21 Sep 2015 18:45:57 +0530
Subject: [PATCH] FOR_UPSTREAM [VPG]: drm/i915: Cleanup latency setting

The Latency values are incorrect and results in link training failures
when 4 lanes are used, hence removing this.

Change-Id: I9ab15614bc0c9320801a018791d3937111c7c482
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-5971
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 9af1337..c22ec3c 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -2976,7 +2976,6 @@ static void chv_pre_enable_dp(struct intel_encoder *encoder)
 		to_intel_crtc(encoder->base.crtc);
 	enum dpio_channel ch = vlv_dport_to_channel(dport);
 	int pipe = intel_crtc->pipe;
-	int data, i;
 	u32 val, stagger;
 
 	mutex_lock(&dev_priv->dpio_lock);
@@ -3007,19 +3006,6 @@ static void chv_pre_enable_dp(struct intel_encoder *encoder)
 	val |= (DPIO_PCS_TX_LANE2_RESET | DPIO_PCS_TX_LANE1_RESET);
 	vlv_dpio_write(dev_priv, pipe, VLV_PCS23_DW0(ch), val);
 
-	/* Program Tx lane latency optimal setting*/
-	for (i = 0; i < 4; i++) {
-		/* Set the latency optimal bit */
-		data = (i == 1) ? 0x0 : 0x6;
-		vlv_dpio_write(dev_priv, pipe, CHV_TX_DW11(ch, i),
-				data << DPIO_FRC_LATENCY_SHFIT);
-
-		/* Set the upar bit */
-		data = (i == 1) ? 0x0 : 0x1;
-		vlv_dpio_write(dev_priv, pipe, CHV_TX_DW14(ch, i),
-				data << DPIO_UPAR_SHIFT);
-	}
-
 	/* Data lane stagger programming */
 	if (intel_crtc->config.port_clock > 270000)
 		stagger = 0x18;
-- 
1.9.1

